cpp/ql/src/Likely Bugs/Underspecified Functions/MistypedFunctionArguments.ql

```diff
@@ -0,0 +1,109 @@
+/**
+ * @name Call to a function with one or more incompatible arguments
+ * @description A call to a function with at least one argument whose type does
+ * not match the type of the corresponding function parameter.  This may indicate
+ * that the author is not familiar with the function being called.  Passing mistyped
+ * arguments on a stack may lead to unpredictable function behavior.
```

Comment:
I think you should delete the second sentence. It might insult users and does nothing to explain what was matched or what the consequences are, which is what a `@description` should summarize. I think you can also delete "on a stack" since in most cases it's no better to pass mistyped arguments in registers. The first sentence also seems to be truncated.

Here's my attempt at rewriting the description to follow https://github.com/Semmle/ql/blob/master/docs/query-metadata-style-guide.md#query-descriptions-description:

`@description When the type of a function argument is not compatible with the type of the corresponding parameter, it may lead to unpredictable behavior.`